blast/querier/server/internal/server/diamond/ep.go

```diff
@@ -0,0 +1,119 @@
+package diamond
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	"github.com/vulpine-io/midl/v1/pkg/midl"
+	"github.com/vulpine-io/split-pipe/v1/pkg/spipe"
+
+	"server/internal/config"
+	"server/internal/mtx"
+	"server/internal/server"
+	"server/internal/server/middleware"
+	"server/internal/util"
+)
+
+func NewDiamondEndpoint(conf config.Config) *Endpoint {
+	return &Endpoint{config: conf}
+}
+
+type Endpoint struct {
+	config config.Config
+}
+
+func (e Endpoint) Handle(req midl.Request) midl.Response {
+	mtx.RecordRequest()
+
+	// Get logger with context.
+	log := req.AdditionalContext()[middleware.KeyLogger].(*logrus.Entry)
+
+	log.Debug("deserializing request")
+	request := new(Request)
+
+	if err := json.Unmarshal(req.Body(), request); err != nil {
+		log.Debug("failed to deserialize request: ", err.Error())
+		return server.NewFailResponse(err.Error())
+	}
+
+	log = log.WithField("job-id", request.JobID)
+
+	workDir := filepath.Join(e.config.OutDir, request.JobID)
+	log.Debug("work directory: ", workDir)
+
+	// Check that the workspace exists
+	if _, err := os.Stat(workDir); err != nil {
+		log.Error("failed to stat workspace directory: ", err.Error())
+		return server.NewFailResponse(err.Error())
+	}
+
+	stdout, err := os.Create(filepath.Join(workDir, "log.txt"))
+	if err != nil {
+		log.Error("failed to create stdout log file: ", err.Error())
+		// Create failed flag
+		_ = util.TouchFailedFlag(workDir)
+		return server.NewFailResponse(err.Error())
+	}
+	defer stdout.Close()
+
+	errFile, err := os.Create(filepath.Join(workDir, "error.txt"))
+	if err != nil {
+		log.Debug("Failed to create stderr log file: ", err.Error())
+		// Create failed flag
+		_ = util.TouchFailedFlag(workDir)
+		return server.NewFailResponse(err.Error())
+	}
+	defer errFile.Close()
+
+	stderr := spipe.NewSplitWriter(errFile, util.StdErrLogger(log))
+	stderr.IgnoreErrors(true)
+
+	args := make([]string, len(request.Params)+5)
+	args[0] = request.Tool
```

Comment:
Should we be checking for malicious values for Tool and Params?  Don't know what valid values look like but seems like we should not just take whatever string the caller sends.

Reply:
In this case, the only caller is the primary legacy multi-blast service.  This service is stack internal, and the value submitted here will be from an enum in Kotlin.